BFS/src/Bridge.java

```diff
@@ -0,0 +1,69 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+
+public class Bridge {
+
+    private static int edge_num, start, goal;
+    private static ArrayList<Integer> x_vals, y_vals;
+
+    private static void start(String file) {
+
+        try {
+            Scanner s = new Scanner(new File(file));
+            s.nextInt();
+            edge_num = s.nextInt();
+            start = s.nextInt();
+            goal = s.nextInt();
+            
+            x_vals = new ArrayList<Integer>();
+            y_vals = new ArrayList<Integer>();
+            
+            for (int i = 0; i < edge_num; i++) {
+                x_vals.add(s.nextInt());
+                y_vals.add(s.nextInt());
+            }
+       
+        } catch (Exception e) {
+            System.out.println("file not opened");
+        }
+
+    }
+
+
+    private static ArrayList<Integer> get_child(int parent) {
+
+        ArrayList<Integer> children = new ArrayList<Integer>();
+        for (int i = 0; i < x_vals.size(); i++) {
+            if (x_vals.get(i) == parent) children.add(y_vals.get(i));
+            if (y_vals.get(i) == parent) children.add(x_vals.get(i));
+        }
+        return children;
+    }
+
+    private static int get_depth(int start, int goal) {
+
+        ArrayList<int[]> queue = new ArrayList<>();
```

Comment:
instead of the arraylist, you should look at APIs for `ArrayDeque` or `LinkedList` which have specific implementations of Stack and Queues like `poll()` and `offer()`. It comes in handy in a lot of times when you run out of ideas.